src/lib/homestore_backend/pg_blob_iterator.cpp

```diff
@@ -36,102 +82,178 @@ PG* HSHomeObject::PGBlobIterator::get_pg_metadata() {
+    std::memcpy(payload.bytes(), builder.GetBufferPointer(), builder.GetSize());
+
+    pack_resync_message(meta_blob, SyncMessageType::SHARD_META, payload);
+    return true;
 }
 
-int64_t HSHomeObject::PGBlobIterator::get_next_blobs(uint64_t max_num_blobs_in_batch, uint64_t max_batch_size_bytes,
-                                                     std::vector< BlobInfoData >& blob_data_vec, bool& end_of_shard) {
-    // end_of_shard = false;
-    // uint64_t total_bytes = 0, num_blobs = 0;
-    // while (true) {
-    //     auto r = home_obj_.query_blobs_in_shard(pg_id_, cur_shard_seq_num_, cur_blob_id_ + 1, max_num_blobs_in_batch);
-    //     if (!r) { return -1; }
-    //     auto& index_results_vec = r.value();
-    //     for (auto& info : index_results_vec) {
-    //         if (info.pbas == HSHomeObject::tombstone_pbas) {
-    //             // Skip deleted blobs
-    //             continue;
-    //         }
-    //         auto result = home_obj_
-    //                           ._get_blob_data(repl_dev_, info.shard_id, info.blob_id, 0 /*start_offset*/,
-    //                                           0 /* req_len */, info.pbas)
-    //                           .get();
-    //         if (!result) {
-    //             LOGE("Failed to retrieve blob for shard={} blob={} pbas={}", info.shard_id, info.blob_id,
-    //                  info.pbas.to_string(), result.error());
-    //             return -1;
-    //         }
-    //
-    //         auto& blob = result.value();
-    //         num_blobs++;
-    //         total_bytes += blob.body.size() + blob.user_key.size();
-    //         if (num_blobs > max_num_blobs_in_batch || total_bytes > max_batch_size_bytes) { return 0; }
-    //
-    //         BlobInfoData blob_data{{info.shard_id, info.blob_id, std::move(info.pbas)}, std::move(blob)};
-    //         blob_data_vec.push_back(std::move(blob_data));
-    //         cur_blob_id_ = info.blob_id;
-    //     }
-    //
-    //     if (index_results_vec.empty()) {
-    //         // We got empty results from index, which means we read
-    //         // all the blobs in the current shard
-    //         end_of_shard = true;
-    //         cur_shard_seq_num_++;
-    //         cur_blob_id_ = -1;
-    //         break;
-    //     }
-    // }
-    //
-    return 0;
+BlobManager::AsyncResult< sisl::io_blob_safe > HSHomeObject::PGBlobIterator::load_blob_data(
+    const BlobInfo& blob_info, ResyncBlobState& state) {
+    auto shard_id = blob_info.shard_id;
+    auto blob_id = blob_info.blob_id;
+    auto blkid = blob_info.pbas;
+    auto const total_size = blob_info.pbas.blk_count() * repl_dev_->get_blk_size();
+    sisl::io_blob_safe read_buf{total_size, io_align};
+
+    sisl::sg_list sgs;
+    sgs.size = total_size;
+    sgs.iovs.emplace_back(iovec{.iov_base = read_buf.bytes(), .iov_len = read_buf.size()});
+
+    LOGD("Blob get request: shard_id={}, blob_id={}, blkid={}", shard_id, blob_id, blkid.to_string());
```

Comment:
why we copy paste the code from ` _get_blob_data`

Reply:
`_get_blob_data` returns `Blob` which doesn't include blobHeader. In addition, we need a different way to treat corrupted blob.